taichi/program/sparse_matrix.cpp

```diff
@@ -276,6 +306,282 @@ void make_sparse_matrix_from_ndarray_cusparse(Program *prog,
 #endif
 }
 
+// Reference::https://docs.nvidia.com/cuda/cusparse/index.html#csrgeam2
+std::unique_ptr<SparseMatrix> CuSparseMatrix::addition(
+    const CuSparseMatrix &other,
+    const float alpha,
+    const float beta) const {
+#if defined(TI_WITH_CUDA)
+  // Get information of this matrix: A
+  size_t nrows_A = 0, ncols_A = 0, nnz_A = 0;
+  void *drow_offsets_A = NULL, *dcol_indices_A = NULL, *dvalues_A = NULL;
+  cusparseIndexType_t csrRowOffsetsType_A, csrColIndType_A;
+  cusparseIndexBase_t idxBase_A;
+  cudaDataType valueType_A;
+  TI_ASSERT(matrix_ != NULL);
```

Comment:
```suggestion
  TI_ASSERT(matrix_ != nullptr);
```
nit: use `nullptr` instead of `NULL`

Reply:
Let's replace all the `NULL`s in this file :-)